### PR TITLE
Data adapter/provider

### PR DESCRIPTION
**Pull request by jrhea**
## PR Description

We need to be able to produce different views of beaconchain data in different views and different formats
- [x] create data adapter to aggregate data to desired level of detail
- [x] create data provider that converts to the desired output format (csv, json, avro, stream, etc)
- [x] produce an example csv file for @dangerousfood to use at ETHParis
## Fixed Issue(s)

resolves some of #430

### CODE REVIEW

File: artemis/src/main/java/tech/pegasys/artemis/Artemis.java [hunk 1]
```diff
@@ -14,53 +14,30 @@
...
-          BeaconNodeService.class,
-          ChainStorageService.class);
-      // Start services
-      ServiceController.startAll(cliArgs);
-    } catch (Exception e) {
-      System.out.println(e.toString());
-    }
+    Security.addProvider(new BouncyCastleProvider());
+    // Process Command Line Args
+    CommandLineArguments cliArgs = new CommandLineArguments();
```
Comment (cemozerr) [inline artemis/src/main/java/tech/pegasys/artemis/Artemis.java:26]: clean

File: artemis/src/main/java/tech/pegasys/artemis/BeaconNode.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import tech.pegasys.artemis.networking.p2p.MockP2PNetwork;
+import tech.pegasys.artemis.networking.p2p.api.P2PNetwork;
+import tech.pegasys.artemis.services.ServiceController;
+import tech.pegasys.artemis.services.beaconchain.BeaconChainService;
+import tech.pegasys.artemis.services.chainstorage.ChainStorageService;
+import tech.pegasys.artemis.services.powchain.PowchainService;
+import tech.pegasys.artemis.util.cli.CommandLineArguments;
+import tech.pegasys.artemis.util.hashtree.HashTreeUtil;
+
+public class BeaconNode {
```
Comment (cemozerr) [inline artemis/src/main/java/tech/pegasys/artemis/BeaconNode.java:40]: I love how nice BeaconNode looks, great job man

**Review by cemozerr — APPROVED** [review]
LGTM

File: artemis/src/main/java/tech/pegasys/artemis/BeaconNode.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  private static final Logger LOG = LogManager.getLogger(CSVProvider.class.getName());
```
Comment (akhila-raju) [inline artemis/src/main/java/tech/pegasys/artemis/BeaconNode.java:41]: hmm I think this is supposed to be indented, did you run spotless? :)

Comment (jrhea) [reply artemis/src/main/java/tech/pegasys/artemis/BeaconNode.java:41]: lemme check...i don't know how you all notice stuff like that.  seriously, makes me think there is something wrong with my brain

File: artemis/src/main/java/tech/pegasys/artemis/BeaconNode.java (hunk 2 above)
Comment (jrhea) [reply artemis/src/main/java/tech/pegasys/artemis/BeaconNode.java:40]: thanks dude

**Review by akhila-raju — APPROVED** [review]
LGTM - just one comment :)

File: artemis/src/main/java/tech/pegasys/artemis/BeaconNode.java (hunk 3 above)
Comment (jrhea) [reply artemis/src/main/java/tech/pegasys/artemis/BeaconNode.java:41]: ok so i ran spotless.....no change

Comment (akhila-raju) [reply artemis/src/main/java/tech/pegasys/artemis/BeaconNode.java:41]: It's not our code! So it's easier to spot 😛

**Pull request merged**